README.md

```diff
@@ -158,35 +158,18 @@ For pure Python projects, OpenAssetIO is available on PyPI, simply run:
 python -m pip install openassetio
 ```
 
-### Build from source (using the ASWF Docker image)
-
-C++ projects require a from-source build, the most convenient way to
-achieve this is to use the [ASWF Docker image](https://github.com/AcademySoftwareFoundation/aswf-docker).
-
-The CY22 image contains all dependencies currently required for building
-OpenAssetIO. For example, to build and install OpenAssetIO (by default
-to a `dist` directory under the build directory) via a container, from
-the repository root run
-
-```shell
-docker run -v `pwd`:/src aswf/ci-base:2022.2 bash -c '
-  cd /src && \
-  cmake -S . -B build && \
-  cmake --build build && \
-  cmake --install build'
-```
-
-The install tree (`dist`) will contain a complete bundle, including the
-core C++ shared library, Python extension module (which dynamically
-links to the core C++ library) and Python sources.
-
-The created bundle is therefore suitable for use in both C++ and Python
-applications.
-
 ### Other build methods
 
 For detailed instructions see [BUILDING](BUILDING.md).
 
+> **Note**
+>
+> Currently, C++ projects require a from-source build.
+> A convenient way to achieve this out of the box is to use a sandboxed
+> build workflow, with the [ASWF Docker image](https://github.com/AcademySoftwareFoundation/aswf-docker).
+>
+> You can find instructions for that build workflow [here](BUILDING.md#sandboxed-builds).
+
```

Comment:
[consider] It feels like the order of information might be a little inverted in this section. Can we add the bit about C++ builds needing a source build before the link to `BUILDING.md`?

eg, open with:

> C++ projects require a from-source build, the most convenient way to achieve this is to use the [ASWF Docker image](BUILDING.md#sandboxed-builds) workflow.
> 
> For detailed build instructions and information on dependency requirements, see [BUILDING](BUILDING.md).

Reply:
I guess I was thinking that keeping the "C++ require you to build manually" thing as a suffix was appropriate due to the hopefully temporary nature of that statement, but I guess there's no reason to alter the layout of the docs based on what's about to happen.

Reply:
Right got you - that makes sense - I suppose which ever way - it might be good to add a little more context for the "other building methods" (eg why you might need them?)

Reply:
Yeah. I've taken a bit of a less-is-more approach in the end. Added a piece of context, but removed as much build specific information from the README as I can, favouring rather to defer out almost entirely to BUILDING, as this section is about installing, not building.

[5f3796a](https://github.com/OpenAssetIO/OpenAssetIO/pull/711/commits/5f3796a7d3cdeaa11c907f6a4f27823484e2200a)

What do you think? We lose the hint to try a containerised build, but i'm not sure how valuable being explicit about that was anymore anyhow, given how much easier the regular build steps are to follow now.

Reply:
(slight wording changes in subsequent fixups)